install/update_0831_084.php

```diff
@@ -2353,31 +2685,37 @@ function updateNetworkFramework(&$ADDTODISPLAYPREF) {
                                       'Update connections between IPAddress and IPNetwork'));
 
    // Here, we are sure that there is only IPv4 addresses. So, the SQL requests are simplified
-   $query = "SELECT `id`, `address_3`, `netmask_3`
-             FROM `glpi_ipnetworks`";
+   $query = [
+      'SELECT' => ["id", "address_3", "netmask_3"],
+      'FROM'   => "glpi_ipnetworks"
+   ];
 
-   if ($network_result = $DB->query($query)) {
+   if ($network_result = $DB->request($query)) {
       unset($query);
-      while ($ipnetwork_row = $DB->fetch_assoc($network_result)) {
+      while ($ipnetwork_row = $network_result->next()) {
          $ipnetworks_id = $ipnetwork_row['id'];
          $netmask       = floatval($ipnetwork_row['netmask_3']);
          $address       = floatval($ipnetwork_row['address_3']) & $netmask;
 
-         $query = "SELECT `id`
-                   FROM `glpi_ipaddresses`
-                   WHERE (`glpi_ipaddresses`.`binary_3` & '$netmask') = $address
-                         AND `glpi_ipaddresses`.`version` = '4'
-                   GROUP BY `items_id`";
-
-         if ($ipaddress_result = $DB->query($query)) {
-            unset($query);
-            while ($link = $DB->fetch_assoc($ipaddress_result)) {
-               $query = "INSERT INTO `glpi_ipaddresses_ipnetworks`
-                                (`ipaddresses_id`, `ipnetworks_id`)
-                         VALUES ('".$link['id']."', '$ipnetworks_id')";
-               $DB->query($query);
-               unset($query);
-            }
+         $ipaddress_result = $DB->request([
+            'SELECT' => "id",
+            'FROM'   => 'glpi_ipaddresses',
+            'WHERE'  => [
+               new \QueryExpression(
+                  "(". DBmysql::quoteName("glpi_ipaddresses.binary_3") . "& " .
+                  DBmysql::quoteValue($netmask) . ")"
```

Comment:
```suggestion
                  DBmysql::quoteValue($netmask) . ")  = " . DBmysql::quoteValue($address)
```

Reply:
You cannot use a QueryExpression as array key, you have to put the `  = " . DBmysql::quoteValue($address)` inside QueryExpression.